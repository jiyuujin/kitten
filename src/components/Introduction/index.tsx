import React, { FC } from 'react'
import * as SC from './index.module.scss'
import { SiteMetadataTypes } from '../../types'

const Introduction: FC<SiteMetadataTypes> = ({ data }) => {
    const positionList: string[] = data.position.split(',')

    return (
        <div className={SC.basic}>
            基本情報
            <div className={SC.description}>{data.biography}</div>
            <div className={SC.subtitle}>職業</div>
            <div className={SC.description}>
                <ul>
                    {positionList.map((p: string) => {
                        return <li key={p}>{p}</li>
                    })}
                </ul>
            </div>
        </div>
    )
}

export default Introduction
