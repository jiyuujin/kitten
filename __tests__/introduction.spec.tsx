import React from 'react'
import renderer from 'react-test-renderer'
import Introduction from '../src/components/Introduction'
import { siteMetadata } from './constants'

jest.mock('react-intl', () => {
    const reactIntl = jest.requireActual('react-intl')
    const intl = reactIntl.createIntl({
        locale: 'en',
    })
    return {
        ...reactIntl,
        useIntl: () => intl,
    }
})

describe('Introduction', () => {
    it('render correctly', () => {
        const tree = renderer
            .create(<Introduction data={siteMetadata} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
