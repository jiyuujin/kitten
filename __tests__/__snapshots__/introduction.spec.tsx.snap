// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction render correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="section"
  >
    <div>
      labels.basic_info
    </div>
    <div>
      basic_biography
    </div>
    <div>
      labels.position
    </div>
    <div>
      <ul>
        <li>
          Front-End Engineer
        </li>
        <li>
          Web Developer
        </li>
        <li>
          HR Adviser
        </li>
      </ul>
    </div>
  </div>
</div>
`;
